### ISSUE TITLE
filename encoding problem

### ISSUE THREAD

**Issue by stray-leone** [evansd/whitenoise#107]
when static file name is non-ascii, error is shown at below code line at web application runtime
https://github.com/evansd/whitenoise/blob/008b938354fc0d4a5cf04eb22a7d494cfc564c73/whitenoise/base.py#L90

error message is beloow

> UnicodeDecodeError: 'ascii' codec can't decode byte 0xeb in position 9: ordinal not in range(128)

could you give me some way of solving this problem??

**Comment by evansd** [comment]
I can't reproduce this issue. I've added a file called `nonascii_✓.txt` to my test suite but WhiteNoise handles it without any problem. Can you provide more details?

It's probably that the line causing the error needs to be changed to use unicode strings i.e.
```python
    url = prefix + os.path.relpath(path, root).replace(u'\\', u'/')
```
But I don't want to make any changes until I can reproduce the issue and understand what's going on.

**Comment by glasslion** [comment]
I am able to reproduce this bug.

Traceback:
```python
UnicodeDecodeError: 'ascii' codec can't decode byte 0xe2 in position 21: ordinal not in range(128)
  File "raven/middleware.py", line 35, in __call__
    iterable = self.application(environ, start_response)
  File "django/core/handlers/wsgi.py", line 158, in __call__
    self.load_middleware()
  File "django/core/handlers/base.py", line 53, in load_middleware
    mw_instance = mw_class()
  File "whitenoise/middleware.py", line 20, in __init__
    super(WhiteNoiseMiddleware, self).__init__(None)
  File "whitenoise/django.py", line 42, in __init__
    self.add_files(self.static_root, prefix=self.static_prefix)
  File "whitenoise/base.py", line 84, in add_files
    self.update_files_dictionary(root, prefix)
  File "whitenoise/base.py", line 90, in update_files_dictionary
    url = prefix + os.path.relpath(path, root).replace('\\', '/')
```

In my case, the type of `prefix` is `unicode` in the left hand while the type of `path` and `root` in the right hand are all `str`. The catenation of a `str` and `unicode` trigger the exception. However this not the root cause.  `root` comes from `WhiteNoise.static_root` which come from  `settings.STATIC_ROOT`.  settings.STATIC_ROOT can be a `str`.

This bug can not be produced in the test suite because `static_root` is guaranteed to be unicode(https://github.com/evansd/whitenoise/blob/master/tests/utils.py#L14). How this assumption probably is not right.

Django doc does not mention whether the `STATIC_ROOT` setting should be `unicode` or `str`. In Python2, this setting generated by the `startproject` command is a `str` . My STATIC_ROOT setting is borrowed from a well-known django project template: [cookiecutter-django](https://github.com/pydanny/cookiecutter-django), which is also a str.

**Comment by glasslion** [comment]
For now, I can get around this issue by converting django's `STATIC_ROOT` setting to `unicode`. I still believe this issue should be fixed in the whitenoise's side either because it is **too dangerous**. This issue will **take down** the **whole** site and broke every request. Besides, it's very hard to notice this issue during developing and testing.

**Comment by evansd** [comment]
Re-opening until I've had a chance to issue a new release which includes this fix

**Comment by evansd** [comment]
Fixed in v3.2.2
